layers/+tags/gtags/funcs.el

```diff
@@ -73,43 +49,20 @@ Otherwise does nothing."
     (helm-gtags-dwim)))
 
 (defun spacemacs/helm-gtags-maybe-dwim ()
-  "Runs `helm-gtags-dwim' if `gtags-enable-by-default' is on.
-Otherwise does nothing."
+  "helm-gtags-dwim in same window"
   (interactive)
-  (when gtags-enable-by-default
-    (call-interactively 'helm-gtags-dwim)))
+  (call-interactively 'helm-gtags-dwim))
 
 (defun spacemacs/helm-gtags-define-keys-for-mode (mode)
-  "Define key bindings for the specific MODE."
-  ;; The functionality of `helm-gtags-mode' is pretty much entirely superseded
-  ;; by `ggtags-mode', so we don't add this hook
-  ;; (let ((hook (intern (format "%S-hook" mode))))
-  ;;   (add-hook hook 'helm-gtags-mode))
+  "Obsolete, does nothing."
+  (message "spacemacs/helm-gtags-define-keys-for-mode does nothing! %s doesn't have to call it anymore."
```

Comment:
Why not to just remove it?

Reply:
Well I want to leave it to the next PR if this one goes through. For example here is the list of the callers:
```
./layers/+lang/javascript/packages.el:75:  (spacemacs/helm-gtags-define-keys-for-mode 'js2-mode))
./layers/+lang/lua/packages.el:81:  (spacemacs/helm-gtags-define-keys-for-mode 'lua-mode))
./layers/+lang/fsharp/packages.el:85:  (spacemacs/helm-gtags-define-keys-for-mode 'fsharp-mode))
./layers/+lang/python/packages.el:160:  (spacemacs/helm-gtags-define-keys-for-mode 'python-mode))
./layers/+lang/latex/packages.el:216:  (spacemacs/helm-gtags-define-keys-for-mode 'latex-mode))
./layers/+lang/windows-scripts/packages.el:92:  (spacemacs/helm-gtags-define-keys-for-mode 'bat-mode))
./layers/+lang/java/packages.el:63:  (spacemacs/helm-gtags-define-keys-for-mode 'java-mode))
./layers/+lang/emacs-lisp/packages.el:309:  (spacemacs/helm-gtags-define-keys-for-mode 'emacs-lisp-mode))
./layers/+lang/kotlin/packages.el:57:  (spacemacs/helm-gtags-define-keys-for-mode 'kotlin-mode))
./layers/+lang/csharp/packages.el:69:  (spacemacs/helm-gtags-define-keys-for-mode 'csharp-mode))
./layers/+lang/asm/packages.el:87:  (spacemacs/helm-gtags-define-keys-for-mode 'asm-mode))
./layers/+lang/rust/packages.el:97:  (spacemacs/helm-gtags-define-keys-for-mode 'rust-mode))
./layers/+lang/common-lisp/packages.el:74:  (spacemacs/helm-gtags-define-keys-for-mode 'common-lisp-mode))
./layers/+lang/racket/packages.el:62:  (spacemacs/helm-gtags-define-keys-for-mode 'racket-mode))
./layers/+lang/ocaml/packages.el:100:  (spacemacs/helm-gtags-define-keys-for-mode 'ocaml-mode))
./layers/+lang/ruby/packages.el:134:    (spacemacs/helm-gtags-define-keys-for-mode mode)))
./layers/+lang/haskell/packages.el:317:  (spacemacs/helm-gtags-define-keys-for-mode 'haskell-mode))
./layers/+lang/c-c++/packages.el:200:    (spacemacs/helm-gtags-define-keys-for-mode mode)))
./layers/+lang/erlang/packages.el:71:  (spacemacs/helm-gtags-define-keys-for-mode 'erlang-mode))
./layers/+lang/go/packages.el:81:  (spacemacs/helm-gtags-define-keys-for-mode 'go-mode))
./layers/+lang/scheme/packages.el:105:  (spacemacs/helm-gtags-define-keys-for-mode 'scheme-mode))
./layers/+lang/elixir/packages.el:194:  (spacemacs/helm-gtags-define-keys-for-mode 'elixir-mode))
./layers/+lang/clojure/packages.el:451:  (spacemacs/helm-gtags-define-keys-for-mode 'clojure-mode))
./layers/+lang/d/packages.el:75:  (spacemacs/helm-gtags-define-keys-for-mode 'd-mode))
./layers/+lang/octave/packages.el:56:  (spacemacs/helm-gtags-define-keys-for-mode 'octave-mode))
./layers/+lang/php/packages.el:66:  (spacemacs/helm-gtags-define-keys-for-mode 'php-mode))
./layers/+lang/scala/packages.el:119:    (spacemacs/helm-gtags-define-keys-for-mode 'scala-mode)))
./layers/+lang/shell-scripts/packages.el:109:  (spacemacs/helm-gtags-define-keys-for-mode 'sh-mode))
./layers/+lang/vimscript/packages.el:72:  (spacemacs/helm-gtags-define-keys-for-mode 'vimrc-mode))
```
Change for each layer, in the next PR:
- Remove ggtags and counsel-gtags from layer-packages
- Remove functions layer/post-init-counsel-gtags and layer/post-init-helm-gtags 

This will be huge and tedious I don't want to waste my time if this current PR rejected.